.github/workflows/get_build_targets.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+readonly TMP_OUTPUT_FILE="tmp.txt"
+readonly QUERY_OUTPUT_FILE="tmp_query.txt"
+
+# This limits the directory that bazel query is going to search under.
+readonly SEARCH_FOLDER="//source/common/..."
+
+set -eu -o pipefail
+
+git fetch https://github.com/envoyproxy/envoy.git master 2>/dev/null
+
+git diff --name-only $ORIGINAL_HEAD FETCH_HEAD | while IFS= read -r line
+do
+  # Only targets under those folders.
+  case "$line" in
+    source/*|include/*)
+      bazel query "rdeps($SEARCH_FOLDER, $line, 1)" 2>/dev/null
+      ;;
+  esac
+done | sort -u
```

Comment:
Does sort have a an option to limit the output? Or can we pipe and limit this to some reasonable number of targets (e.g. 10?)

Reply:
and which 10 doesn't really matter I suppose?

Reply:
Yes, also I wonder if we should restrict rdeps to source/common/, i expect that rdeps will probably give a lot.

Reply:
I think it already does that on L7 and L18

Reply:
Oof, thanks. I mixed up the line filter with the search_folder. thanks!